sys/net/application_layer/unicoap/options.c

```diff
@@ -0,0 +1,928 @@
+/*
+ * Copyright (C) 2024-2025 Carl Seifert
+ * Copyright (C) 2024-2025 TU Dresden
+ *
+ * This file is subject to the terms and conditions of the GNU Lesser General
+ * Public License v2.1. See the file LICENSE in the top level directory for
+ * more details.
+ */
+
+/**
+ * @file
+ * @ingroup net_unicoap_options
+ * @brief   Options implementation
+ * @author  Carl Seifert <carl.seifert1@mailbox.tu-dresden.de>
+ */
+
+#include <stdio.h>
+#include <stdint.h>
+#include <errno.h>
+#include <assert.h>
+
+#include "byteorder.h"
+#include "compiler_hints.h"
+
+#include "net/unicoap/message.h"
+#include "net/unicoap/options.h"
+
+#define ENABLE_DEBUG CONFIG_UNICOAP_DEBUG_LOGGING
+#include "debug.h"
+#include "private.h"
+
+#ifndef DOXYGEN
+/**
+ * # Terminology
+ * @note For improved readibility and consistent naming throughout the
+ * option parser, the following terminology is used.
+ *
+ * ## Quantities
+ * - `size`: Number of bytes a given value consists of. `length`is not used due to ambiguity.
+ * - `capacity`: Maximum number of bytes available in a given buffer. A buffer usually has a capacity and the value stored inside a size.
+ * - `count`: A discrete number of elements in a container.
+ * - `width`: Constant number of bits and bytes a certain type occupies. E.g., `uint16_t` has a width of 16 bits or 2 bytes.
+ *
+ * ## Option Format
+ * - `DELTA`: Positive option number difference to the preceding option, zero if first option.
+ * - `GAMMA`: Size of option body attached. RFC 7252 refers to this value as the *Option Length*.
```

Comment:
Option value length?

Reply:
See also https://github.com/RIOT-OS/RIOT/pull/21390#discussion_r2039527152 and https://github.com/RIOT-OS/RIOT/pull/21390#discussion_r2039530279